lib/view_models/page_view_model/join_organization_viewModel.dart

```diff
@@ -0,0 +1,358 @@
+import 'package:talawa/controllers/auth_controller.dart';
+import 'package:talawa/services/preferences.dart';
+import 'package:talawa/services/queries_.dart';
+import 'package:talawa/utils/globals.dart';
+import 'package:talawa/utils/gql_client.dart';
+import 'package:talawa/view_models/base_model.dart';
+
+import '../../locator.dart';
+
+class JoinOrgnizationViewModel extends BaseModel {
+  final Queries _query = Queries();
+  final Preferences _pref = Preferences();
+  String token;
+  String _itemIndex;
+  GraphQLConfiguration graphQLConfiguration = locator<GraphQLConfiguration>();
+  FToast fToast;
+  List _organizationInfo = [];
+  final List _filteredOrgInfo = [];
+  List joinedOrg = [];
+  final AuthController _authController = AuthController();
+  String _isPublic;
+  final String _fetchingOrgError = "";
+  final bool _hasFetchingOrgError = false;
+  final String _sendingRequestToOrgError = "";
+  String _currentUserId = "";
+  List joinedOrganizations = [];
+  List joinedOrganizationsIds = [];
+  bool _isLoaderActive = false;
+  int _loadingIndex = -1;
+
+  bool disposed = false;
+
+  String get currentUserId => _currentUserId;
+  String get fetchingOrgError => _fetchingOrgError;
+  bool get hasFetchingOrgError => _hasFetchingOrgError;
+  String get sendingRequestToOrgError => _sendingRequestToOrgError;
+
+  bool get isLoaderActive => _isLoaderActive;
+  int get loadingIndex => _loadingIndex;
+
+  List get organizationInfo => _organizationInfo;
+  List get filteredOrgInfo => _filteredOrgInfo;
+
+  String get isPublic => _isPublic;
+  String get itemIndex => _itemIndex;
+
+  void initialise(BuildContext context) {
+    fToast = FToast();
+    fToast.init(context);
+    fetchOrg();
+  }
+
+  void setItemIndex(String itemIdx) {
+    _itemIndex = itemIdx;
+    notifyListeners();
+  }
+
+  void setIsPublic(String str) {
+    _isPublic = str;
+    notifyListeners();
+  }
+
+  Widget showError(String msg) {
+    return Center(
+      child: Text(
+        msg,
+        style: const TextStyle(fontSize: 16),
+        textAlign: TextAlign.center,
+      ),
+    );
+  }
+
+  void confirmOrgDialog(String orgName, int index, BuildContext context) {
+    //this is the pop up shown when the confirmation is required
+    if (Platform.isIOS) {
+      showCupertinoDialog(
+        context: context,
+        useRootNavigator: false,
+        builder: (_) => CupertinoAlertDialog(
+          title: const Text("Confirmation"),
+          content:
+              const Text("Are you sure you want to join this organization?"),
+          actions: <Widget>[
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+              child: const Text("Close"),
+            ),
+            TextButton(
+              onPressed: () async {
+                _loadingIndex = index;
+                _isLoaderActive = true;
+                notifyListeners();
```

Comment:
why is a notifyListeners here?

Reply:
So that we can notify the UI file "join_organization.dart" that we are joining or have joined the organization.

Please check 170 line of "join_organization.dart",
  ` model.isLoaderActive == true && model.loadingIndex == index`